.github/workflows/steam.yml

```diff
@@ -0,0 +1,212 @@
+            MAC_ASSET_URL=$(jq -r '.[] | select(.name|test(".*.dmg")) .browser_download_url' <<< "$ASSETS")
+            TYPE="release"
+        else
+            BRANCH=${{ env.STEAM_NIGHTLY_BRANCH }}
+            BUILDS=$(curl -s "${{ github.api_url }}/repos/obsproject/obs-studio/actions/workflows/${{ env.WORKFLOW_ID }}/runs?per_page=1&event=push&status=success&branch=${{ env.GIT_NIGHTLY_BRANCH }}")
+            ARTIFACTS_URL=$(jq -r '.workflow_runs[].artifacts_url' <<< "$BUILDS")
+            DESC="g$(jq -r '.workflow_runs[].head_sha' <<< "$BUILDS" | cut -c1-9)"
+
+            ARTIFACTS=$(curl -s $ARTIFACTS_URL)
+            WIN_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*win64")) .archive_download_url' <<< "$ARTIFACTS")
+            MAC_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*macos-x86_64")) .archive_download_url' <<< "$ARTIFACTS")
+            TYPE="nightly"
+        fi
+        
+        # Apply overrides from workflow_dispatch
+        if [[ $EVENT == "workflow_dispatch" ]]; then
+            if [[ ! -z "${{ github.event.inputs.win_url_override }}" ]]; then
+                WIN_ASSET_URL="${{ github.event.inputs.win_url_override }}"
+            fi
+            
+            if [[ ! -z "${{ github.event.inputs.mac_url_override }}" ]]; then
+                MAC_ASSET_URL="${{ github.event.inputs.mac_url_override }}"
+            fi
+        fi
+        
+        if [[ -z $WIN_ASSET_URL || -z $MAC_ASSET_URL ]]; then
+            echo "Missing at least one asset URL!"
+            exit 1
+        fi
+        
+        # set env variables for subsequent steps
+        echo "UPLOAD_TYPE=${TYPE}" >> $GITHUB_ENV
+        echo "BRANCH=${BRANCH}" >> $GITHUB_ENV
+        echo "DESC=${DESC}" >> $GITHUB_ENV
+        echo "WIN_ARTIFACT_URL=${WIN_ASSET_URL}" >> $GITHUB_ENV
+        echo "MAC_x86_ARTIFACT_URL=${MAC_ASSET_URL}" >> $GITHUB_ENV
+
+    - name: Restore build cache
+      id: cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/steam/build
+        key: ${{ env.BRANCH }}-${{ env.DESC }}
+        # Using "restore-keys" will restore the most recent cache for the branch, even if the exact cache doesn't exist.
+        # This doesn't set cache-hit to true so it won't skip the upload for nightlies.
+        restore-keys: ${{ env.BRANCH }}
+
+    - name: Determine if Steam upload should run
+      # If the nightly build has already been uploaded and thus a cache exists skip this and the following steps.
+      # Steam does not prevent us from uploading duplicate builds so this would just pollute the dashboard.
+      # This is a bit of a hack and can fail to work if our cache has been evicted or we somehow have no commits for 7 days,
+      # but it's better than nothing!
+      id: should-run
+      run: |
+        SHOULD_RUN=$(if [[ "${{ env.UPLOAD_TYPE }}" == "release" || "${{ steps.cache.outputs.cache-hit }}" != "true" ]]; then echo "true"; else echo "false"; fi)
```

Comment:
You can `echo` the `set-output` part directly within the branch, no need to store it in an intermediary variable.